File: Backend/YılmazMotorOtomotivWebApi/YılmazMotorWeb.Entities/Concretes/Ticket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using YılmazMotorWeb.Entities.Identity;

namespace YılmazMotorWeb.Entities.Concretes
{
	public class Ticket
	{
		public int Id { get; set; }
		public string Subject { get; set; }
		public string Message { get; set; }
		public DateTime CreatedAt { get; set; } = DateTime.Now;
		public int UserId { get; set; }
		public TicketStatus Status { get; set; } = TicketStatus.Open;
		public AppUser User { get; set; }
		public ICollection<TicketReply> Replies { get; set; }
	}
}

File: Frontend/yilmazMotorOtomotiv/src/app/Models/ticket.ts
import { UserDto } from "./userDto";

export interface TicketDto {
  id: number;
  subject: string;
  message: string;
  createdAt: Date;
  userId: number;
  status: TicketStatus;
  user?: UserDto;
  replies?: TicketReplyDto[];
}

export interface TicketReplyDto {
  id: number;
  ticketId: number;
  replyMessage: string;
  repliedAt: Date;
  userId: number;
  user?: UserDto;
}

export interface CreateTicketDto {
  subject: string;
  message: string;
  userId: number;
}

export interface CreateTicketReplyDto {
  ticketId: number;
  replyMessage: string;
  userId: number;
}

export enum TicketStatus {
  Open = 0,
  Answered = 1,
  Closed = 2
}
